### PR TITLE
Fix `Any` with `float` in `_TreeNode.children`

### PR DESCRIPTION
**Pull request by aanghelidi**
## Motivation

Fix typing of `_TreeNode.children` in `_brute_force.py` as described in #5034 . Additional changes have been applied to keep type hinting consistent.

## Description of the changes

* Change `Optional[Dict[Any, "_TreeNode"]]` to `Optional[Dict[float, "_TreeNode"]]`
* Change **search_space** in `expand` to `Iterable[float]`3
* Change **params_and_search_spaces** in `add_path` to `Iterable[Tuple[str, Iterable[float], float]] `
* Change return type of `sample_child` to `float`
* Change return type of `_enumerate_candidates` to `Sequence[float]`

Changes have been checked with `./formats.sh`.

Resolves: #5034

### CODE REVIEW

**Comment by contramundum53** [comment]
Could you also change 
https://github.com/optuna/optuna/pull/5040/files#diff-f2a9050c6a6aa97c3cacdd1edbf2e46f8e64ebbb317bab29f544c2d7aecb2ed9R148
 and 
 https://github.com/optuna/optuna/pull/5040/files#diff-f2a9050c6a6aa97c3cacdd1edbf2e46f8e64ebbb317bab29f544c2d7aecb2ed9L243
 ?

File: optuna/samplers/_brute_force.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def set_leaf(self) -> None:
         self.expand(None, [])
 
     def add_path(
-        self, params_and_search_spaces: Iterable[Tuple[str, Iterable[Any], Any]]
+        self, params_and_search_spaces: Iterable[Tuple[str, Iterable[float], Any]]
```
Comment (contramundum53) [inline optuna/samplers/_brute_force.py:57]:
```suggestion
        self, params_and_search_spaces: Iterable[Tuple[str, Iterable[float], float]]
```
The value is also in internal representation.

**Comment by contramundum53** [comment]
@nabenabe0928 Could you review this PR?

**Comment by nabenabe0928** [comment]
> @nabenabe0928 Could you review this PR?

Yes

**Comment by nabenabe0928** [comment]
Apparently, categorical parameters' `internal_repr` is an integer, but not a float number.

Example here:

```python
import optuna


def f(trial: optuna.Trial):
    a = trial.suggest_int("a", 1, 3)
    c = int(trial.suggest_categorical("c", ["1", "2", "3"]))
    return a + c

study = optuna.create_study(sampler=optuna.samplers.BruteForceSampler())
study.optimize(f)
```

**Comment by contramundum53** [comment]
@nabenabe0928 This is intended, as the internal representation in DB is float.

File: optuna/samplers/_brute_force.py (hunk 1 above)
Comment (aanghelidi) [reply optuna/samplers/_brute_force.py]: Thank you for the suggestion @contramundum53 , it has been corrected in my last commit :+1:

**Comment by aanghelidi** [comment]
The last commit fix applies the suggestions mentioned above. Let me know if it is ok now.

**Review by nabenabe0928 — APPROVED** [review]
It seems all the lines that should be modified were correctly changed unless I missed")
Thank you for your work, @aanghelidi

**Comment by aanghelidi** [comment]
Thank you for the review, suggestions and your time @contramundum53, @nabenabe0928 :+1:

**Review by contramundum53 — APPROVED** [review]
LGTM! Thanks for your PR!

**Pull request merged**